fl4health/preprocessing/autoencoders/vae_dim_reduction.py

```diff
@@ -0,0 +1,76 @@
+from abc import ABC
+from pathlib import Path
+
+import torch
+
+
+class Processing(ABC):
+    """
+    Abstact class to add processing methods.
+    """
+
+    def __init__(
+        self,
+        checkpointing_path: Path,
+    ) -> None:
+        self.checkpointing_path = checkpointing_path
+        self.DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        self.load_autoencoder()
+
+    def load_autoencoder(self) -> None:
+        autoencoder = torch.load(self.checkpointing_path)
+        autoencoder.eval()
+        self.autoencoder = autoencoder.to(self.DEVICE)
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}"
+
+
+class AEProcessor(Processing):
+    """Transformer processor to encode the data using basic autoencoder."""
+
+    def __init__(
+        self,
+        checkpointing_path: Path,
+    ) -> None:
+        super().__init__(checkpointing_path)
+
+    def __call__(self, sample: torch.Tensor) -> torch.Tensor:
+        # This transformer is called for the input samples after they are transfered into torch tensors.
+        embedding_vector = self.autoencoder.encode(sample.to(self.DEVICE))
+        return embedding_vector.clone().detach()
+
+
+class VAEProcessor(Processing):
+    """Transformer processor to encode the data using VAE encoder."""
+
+    def __init__(
```

Comment:
Perhaps another option here that would be useful is to have an argument for including both the mean and variance vectors (concatenated) in the projection? Something like `include_variance: bool = False` and the call would then have 
```
mu, logvar = self.autoencoder.encode(sample.to(self.DEVICE))
if self.include_variance:
    return torch.cat(mu.clone().detach(), logvar.clone().detach())
return mu.clone().detach()
```
I'd also just make sure that `torch.cat` would concatenate in the right dimension in a test (this is just off the top of my head 😂 ). This could also be done for the CVAE processor below.